### PR TITLE
[WIP] feat(bindings): add createImgIcon factory

### PR DESCRIPTION
**Pull request by mnajdova**
image

### CODE REVIEW

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    const { size = 16, sizeModifier, ...rest } = props;
+    let height = size;
+    let width = size;
+    let svgUrl = `${baseUrl}/${size}${
+      sizeModifier && type === 'png' ? `_${sizeModifier}` : ''
+    }/${name}.${type}?${refreshUrl}`;
+
+    if (sizeModifier) {
+      const multiplier = sizeModifier.substr(0, sizeModifier.length - 1);
+      height = size * +multiplier;
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]: Should the `sizeModifier` affect the `height` and `width`?

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    // SVGs scale well, so you can generally use the default image.
+    // 1.5x is a special case where both SVGs and PNGs need a different image.
+    // Remove if statements when missing image files for 20_1.5x are provided.
+    if (size === 20 && sizeModifier === '1.5x') {
+      svgUrl = `${baseUrl}/${size}_${sizeModifier}/${name}.${type}?${refreshUrl}`;
+    }
+
+    return React.createElement('img', { src: svgUrl, height, width, ...rest });
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]: Is it ok to set the `width` and `height` here?

File: packages/fluentui/react-bindings/src/icons/types.ts [hunk 3]
```diff
@@ -60,3 +60,13 @@ export type SvgIconCreateFnParams<TProps> = {
   displayName: string;
   handledProps?: (keyof TProps)[];
 };
+
+export type ImgIconProps = { size?: number; sizeModifier?: '1.5x' | '2x' | '3x' | '4x' };
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/icons/types.ts]: What props do we actually want to support here? Can we decide on common icon props for both libraries? @dzearing

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { ImgIconCreateFnParams, ImgIconProps } from './types';
+
+const createImgIcon = ({
+  name,
+  displayName,
+  type = 'svg',
+  baseUrl = 'https://spoprod-a.akamaihd.net/files/fabric/assets/item-types',
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]: Read this from context?

File: packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx [hunk 5]
```diff
@@ -1,13 +1,66 @@
...
+        <AccdbIcon />
+        <AccdbIcon size={20} />
+        <AccdbIcon size={24} />
+        <AccdbIcon size={32} />
+        <AccdbIcon size={40} />
+        <AccdbIcon size={48} />
+        <AccdbIcon size={64} />
+        <AccdbIcon size={96} />
+      </Flex>
+      <Text content={'PNG icons'} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx]:
```suggestion
      <Text content="PNG icons" />
```

nit

File: packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx [hunk 6]
```diff
@@ -1,13 +1,66 @@
...
+        <AccdbPngIcon />
+        <AccdbPngIcon size={20} />
+        <AccdbPngIcon size={24} />
+        <AccdbPngIcon size={32} />
+        <AccdbPngIcon size={40} />
+        <AccdbPngIcon size={48} />
+        <AccdbPngIcon size={64} />
+        <AccdbPngIcon size={96} />
+      </Flex>
+      <Text content={'Size modifiers for size 24 svg'} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx]:
```suggestion
      <Text content="Size modifiers for size 24 svg" />
```

File: packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx [hunk 7]
```diff
@@ -1,13 +1,66 @@
...
+      <Flex gap="gap.smaller">
+        <AccdbIcon size={24} sizeModifier="1.5x" />
+        <AccdbIcon size={24} sizeModifier="2x" />
+        <AccdbIcon size={24} sizeModifier="3x" />
+        <AccdbIcon size={24} sizeModifier="4x" />
+      </Flex>
+      <Text content={'Size modifiers for size 24 png'} />
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Icon/Types/IconExample.shorthand.tsx]:
```suggestion
      <Text content="Size modifiers for size 24 png" />
```

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { ImgIconCreateFnParams, ImgIconProps } from './types';
+import { ImgIconContext } from './imgIconContext';
+
+const defaultUrlResolver = (
+  urlConfig: { [key: string]: any } | undefined,
+  name: string,
+  props: { [key: string]: any },
```
Comment (layershifter) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]: Better typings?

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+) => {
+  const { baseUrl, refreshUrl } = urlConfig || {};
+  const { size = 16, sizeModifier, type } = props || {};
+
+  let svgUrl = `${baseUrl}/${size}${
+    sizeModifier && type === 'png' ? `_${sizeModifier}` : ''
+  }/${name}.${type}?${refreshUrl}`;
```
Comment (layershifter) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]:
```suggestion
  }/${name}.${type}?${queryParam}`;
```

`queryParam` or `search` will be more universal

File: packages/fluentui/react-bindings/src/icons/createImgIcon.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+      const multiplier = sizeModifier.substr(0, sizeModifier.length - 1);
+      height = size * +multiplier;
+      width = size * +multiplier;
+    }
+
+    const urlResolver = imgContextValue?.urlResolver || defaultUrlResolver;
+
+    const svgUrl = urlResolver(imgContextValue?.urlConfig, name, { ...props, type });
+
+    return React.createElement('img', { src: svgUrl, height, width, ...rest });
```
Comment (layershifter) [inline packages/fluentui/react-bindings/src/icons/createImgIcon.ts]: I think that here we have missing accessibility attributes

**Comment by mnajdova** [comment]
Closing this one, as after offline discussions we decided not to change at this moment how the file type icons work.

**Pull request closed without merging**